src/modules/Dashboard/components/ActivityGrid/ActivityGrid.hooks.tsx

```diff
@@ -48,9 +50,18 @@ export const useActivityGrid = (
     () => ({
       editActivity: ({ context }: MenuActionProps<ActivityActionProps>) => {
         const { activityId } = context || {};
+        if (!activityId || !appletId) return;
+        const activity = getActivityById(activityId);
+
+        const navigateTo =
+          activity?.isPerformanceTask && activity?.performanceTaskType
+            ? getPerformanceTaskPath(
+                activity?.performanceTaskType as unknown as EditablePerformanceTasksType,
```

Comment:
I wasn't thrilled to see this `as unknown as EditablePerformanceTasksType` coercion, so I decided to investigate why it was necessary. Turns out it's used in [the Builder's Activities.tsk](https://github.com/ChildMindInstitute/mindlogger-admin/blob/fix/M2-7091-task-urls/src/modules/Builder/features/Activities/Activities.tsx#L68) because only certain tasks are editable – not the A/B Trails ones:

https://github.com/ChildMindInstitute/mindlogger-admin/assets/1282772/f670fa58-91c8-409c-9a58-f18b5dcda673

We're not doing that check here and showing the Edit Activity menu item regardless, and when you try to edit one of those performance task types from the Activity Grid, nothing happens. So we should hide that menu item in these cases.

Further, maybe a comment would good here to explain why this is needed, so future devs who are as picky as me about forced coercion aren't spending time investigating. 😛

Reply:
Added validation to remove edit button from ABTrails tasks
![CleanShot 2024-06-24 at 10 42 03](https://github.com/ChildMindInstitute/mindlogger-admin/assets/2916156/706a9e26-2aa4-416d-be68-54ff925ce767)